packages/image/src/Image.svelte

```diff
@@ -0,0 +1,180 @@
+<script context="module">
+	import { providers, domains, device_sizes, image_sizes } from '__svelte-image-options__.js';
+	import { DEV } from 'esm-env';
+
+	const all_sizes = image_sizes.concat(device_sizes);
+
+	/**
+	 * Calculate the widths to use for the srcset using the following logic:
+	 * - If the sizes prop is set, use the vw sizes in the sizes prop.
+	 * 	 Calculate the smallest ratio to the device size and use that to filter out all sizes that will never be used.
+	 * - Always filter out all sizes above 2x the width (because upscaling beyond 2x is wasteful)
```

Comment:
I think we should avoid ever upscaling. The browser will automatically upscale if needed. All we'd be doing by upscaling is wasting bandwidth.

https://imagekit.io/blog/how-to-resize-image-in-html/ says:
> Never upscale a raster image i.e. JPEG, PNG, WebP, or AVIF images, should never be upscaled as it will result in a blurred output.

Reply:
+1

From vite-imagetools code comments:
>     // the fallback should be the largest image in the fallback format
>     // we assume users should never upsize an image because that is just wasted
>     // bytes since the browser can upsize just as well
https://github.com/JonasKruckenberg/imagetools/blob/82a810b78c3d5b7b44a1dfa0518c397c76f23522/packages/core/src/output-formats.ts#L97-L99